### ISSUE TITLE
Adding suggestions Takes time and hangs the app

### ISSUE THREAD

**Issue by SahajRana** [Mauker1/MaterialSearchView#39]
Hey! I have been using your View For implementation but when i add suggestions in onResume() , it hangs the app for 3-4 seconds while going in onResume() lifecycle.

**Comment by Mauker1** [comment]
How big is your suggestion list?

**Comment by SahajRana** [comment]
Hey! could u find any solution to it..? thanks!

**Comment by Mauker1** [comment]
This is happening because on the `addSuggestions()` method, there's this line:

```java
mContext.getContentResolver().bulkInsert(
                HistoryContract.HistoryEntry.CONTENT_URI,
                values
        );
```

Which is doing a bulk insert of all your data on the database at once. And even before that, it's looping through your suggestions to make the ContentValues objects which will be inserted.

Perhaps one way of solving this issue is to do this operation on a background thread. But it will require some more tests. I'll keep you posted.

**Comment by SahajRana** [comment]
okie.. I think that would be a great Idea. Thanks!

**Comment by SahajRana** [comment]
hey! Did u find a way of background Treading it.. Thanks!

**Comment by Mauker1** [comment]
Not yet, as seen on issue #11 , we might change the content provider so for now I'm not tackling this issue.

**Comment by Mauker1** [comment]
The alpha version of V 2.0.0 might've already solved this issue, but I need to test it with a large dataset.

**Comment by Mauker1** [comment]
Tested with a large dataset (100k items). The app didn't hang. This will be available on 2.0.0-beta01.
